hw/application_fpga/fw/tk1_mem.h

```diff
@@ -7,96 +7,125 @@
-
-    TK1_MMIO_TOUCH_STATUS     = TK1_MMIO_TOUCH_BASE | 0x24,
-    TK1_MMIO_TOUCH_STATUS_EVENT_BIT = 0,
-
-    // This will only ever exist in QEMU:
-    TK1_MMIO_QEMU_DEBUG       = TK1_MMIO_QEMU_BASE | 0x1000,
-
-    TK1_MMIO_TK1_NAME0        = TK1_MMIO_TK1_BASE | TK1_NAME0_SUFFIX,
-    TK1_MMIO_TK1_NAME1        = TK1_MMIO_TK1_BASE | TK1_NAME1_SUFFIX,
-    TK1_MMIO_TK1_VERSION      = TK1_MMIO_TK1_BASE | TK1_VERSION_SUFFIX,
-    TK1_MMIO_TK1_SWITCH_APP   = TK1_MMIO_TK1_BASE | 0x20,
-    TK1_MMIO_TK1_LED          = TK1_MMIO_TK1_BASE | 0x24,
-    TK1_MMIO_TK1_LED_R_BIT    = 2,
-    TK1_MMIO_TK1_LED_G_BIT    = 1,
-    TK1_MMIO_TK1_LED_B_BIT    = 0,
-    TK1_MMIO_TK1_GPIO         = TK1_MMIO_TK1_BASE | 0x28,
-    TK1_MMIO_TK1_GPIO1_BIT    = 0,
-    TK1_MMIO_TK1_GPIO2_BIT    = 1,
-    TK1_MMIO_TK1_GPIO3_BIT    = 2,
-    TK1_MMIO_TK1_GPIO4_BIT    = 3,
-    TK1_MMIO_TK1_APP_ADDR     = TK1_MMIO_TK1_BASE | 0x30,
-    TK1_MMIO_TK1_APP_SIZE     = TK1_MMIO_TK1_BASE | 0x34,
-    TK1_MMIO_TK1_BLAKE2S      = TK1_MMIO_TK1_BASE | 0x40,
-    TK1_MMIO_TK1_CDI_FIRST    = TK1_MMIO_TK1_BASE | 0x80,
-    TK1_MMIO_TK1_CDI_LAST     = TK1_MMIO_TK1_BASE | 0x9c, // Address of last 32-bit word of CDI.
-    TK1_MMIO_TK1_UDI_FIRST    = TK1_MMIO_TK1_BASE | 0xc0,
-    TK1_MMIO_TK1_UDI_LAST     = TK1_MMIO_TK1_BASE | 0xc4, // Address of last 32-bit word of UDI.
-    TK1_MMIO_TK1_RAM_ASLR     = TK1_MMIO_TK1_BASE | 0x100,
-    TK1_MMIO_TK1_RAM_SCRAMBLE = TK1_MMIO_TK1_BASE | 0x104,
-    TK1_MMIO_TK1_CPU_MON_CTRL = TK1_MMIO_TK1_BASE | 0x180,
-    TK1_MMIO_TK1_CPU_MON_FIRST = TK1_MMIO_TK1_BASE | 0x184,
-    TK1_MMIO_TK1_CPU_MON_LAST = TK1_MMIO_TK1_BASE | 0x188,
-};
+#ifndef TKEY_TK1_MEM_H
+#define TKEY_TK1_MEM_H
 
+/*
+
+  The canonical location of this file is in:
+
+  https://github.com/tillitis/tillitis-key1
+
+  /hw/application_fpga/fw/tk1_mem.h
+
+  The contents are derived from the Verilog code. For use by QEMU model,
+  firmware, and apps.
+
+  Memory map
+
+  Top level prefix, the first 2 bits in a 32-bit address:
+
+  name	prefix	address length
+  --------------------------------------------------------
+  ROM		0b00	30 bit address
+  RAM		0b01	30 bit address
+  Reserved	0b10
+  MMIO		0b11	6 bits for core select, 24 bits rest
+
+  Address Prefix, the first 8 bits in a 32-bit address:
+
+  name		prefix
+  --------------------
+  ROM		0x00
+  RAM		0x40
+  MMIO		0xc0
+  MMIO TRNG	0xc0
+  MMIO TIMER	0xc1
+  MMIO UDS	0xc2
+  MMIO UART	0xc3
+  MMIO TOUCH	0xc4
+  MMIO FW_RAM	0xd0
+  MMIO TK1	0xff
+ */
+
+#define TK1_ROM_BASE 0x00000000
+#define TK1_RAM_BASE 0x40000000
+#define TK1_RAM_SIZE 0x20000
+
+#define TK1_APP_MAX_SIZE 0x20000
+
+#define TK1_MMIO_FW_RAM_BASE 0xd0000000
+// FW_RAM is 2048 bytes
+#define TK1_MMIO_FW_RAM_SIZE 0x800
+
+#define TK1_MMIO_TRNG_BASE 0xc0000000
+#define TK1_MMIO_TRNG_STATUS 0xc0000024
+#define TK1_MMIO_TRNG_STATUS_READY_BIT 0
+#define TK1_MMIO_TRNG_ENTROPY 0xc0000080
+
+#define TK1_MMIO_TIMER_CTRL 0xc1000020
```

Comment:
`TK1_MMIO_TIMER_BASE` does not exist. 
It is not explicit needed IMO, but we do have the `*_BASE` for all others.